src/bluefinClient.ts

```diff
@@ -1538,6 +1540,298 @@ export class BluefinClient {
     return this.contractCalls.withdrawAllFromMarginBankContractCall();
   };
 
+  closeAllDelistedPositionsAndWithdrawMargin = async (args?: {
+    gasBudget?: number;
+    sponsor?: boolean;
+    dryRunOnly?: boolean;
+  }) => {
+    try {
+      // get exchange info of all markets to read the status
+      const exchangeInfo = await this.getExchangeInfo();
+      const delistedMarkets = filterDelistedMarkets(exchangeInfo);
+
+      if (delistedMarkets.length <= 0) {
+        return {
+          ok: false,
+          code: 400,
+          data: "",
+          message: "No delisted markets",
+        };
+      }
```

Comment:
Specifically, I'd make this check for <= 15 so that users can only do it once all markets are delisted. Otherwise, if someone runs it when some markets are still active it'd mess up the flow

Reply:
Also if you do change it for the above, let's also update the error message